server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/RawDataAggregationOperator.java

```diff
@@ -50,11 +56,21 @@ public RawDataAggregationOperator(
       boolean ascending,
       long maxReturnSize) {
     super(operatorContext, aggregators, child, ascending, timeRangeIterator, maxReturnSize);
+    this.windowManager = new TimeWindowManager(timeRangeIterator);
+    this.finish = false;
+  }
+
+  @Override
+  public boolean hasNext() {
+    if (finish) {
+      return false;
+    }
+    return inputTsBlock != null || child.hasNext() || windowManager.hasNext();
```

Comment:
```suggestion
    return IWindow.hasNext(boolean hasMoreData);
```
You should add a new method in `IWindow` and put these different judgement logics into each different IWindow implementation.